File: 프로그래머스/Level1/크레인인형뽑기.py
def solution(board, moves):
    answer = 0
    stack = [-1]
    size = len(board[0])
    
    for move in moves:
        for i in range(size):
            temp = board[i][move-1]
            if not temp == 0:
                stack.append(temp)
                board[i][move-1] = 0
                
                if len(stack) >= 3 and (stack[-1]==stack[-2]):
                    stack.pop()
                    stack.pop()
                    answer += 1
                break
        
            
    return answer*2

File: swift/프로그래머스/크레인 인형뽑기.swift
import Foundation

func solution(_ board:[[Int]], _ moves:[Int]) -> Int {
    var answer = 0
    let N = board.count
    var boardStack = [[Int]](repeating: [Int](), count: N)
    var mainStack = [Int]()
    
    for i in (0..<N).reversed() {
        for j in 0..<N {
            if board[i][j] == 0 { continue }
            boardStack[j].append(board[i][j])
        }
    }
    
    for move in moves {
        let m = move - 1
        if !boardStack[m].isEmpty {
            let poped = boardStack[m].popLast()!
            if let top = mainStack.last, top == poped {
                let _ = mainStack.popLast()
                answer += 2
            }
            else {
                mainStack.append(poped)
            }
        }
    }
    
    return answer
}
